### PR TITLE
Update network switch procedure and separate simulator calls from real device calls

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Please do not merge this PR before https://github.com/appium/io.appium.settings/pull/7 is merged and published

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
@vrunoa FYI

File: lib/tools/adb-commands.js [hunk 1]
```diff
@@ -390,8 +391,13 @@ methods.isWifiOn = async function () {
 /*
  * on: true (to turn on) or false (to turn off)
  */
-methods.setWifiState = async function (on) {
-  await this.shell(['settings', 'put', 'global', 'wifi_on', on ? 1 : 0]);
+methods.setWifiState = async function (on, isSimulator = false) {
+  if (isSimulator) {
```
Comment (vrunoa) [inline lib/tools/adb-commands.js]: @mykola-mokhnach could this _var_ name be changed to `isEmulator` since android are emulators not simulators

**Comment by mykola-mokhnach** [comment]
No, this is called from android driver, where we just verify whether _opts.avd_ is set or not

**Comment by vrunoa** [comment]
Awesome. What else is set with _io.appium.settings_ app, mock locations ? We can also do that with *adb* and maybe avoid using this app for emulators. I'll create a PR

**Comment by mykola-mokhnach** [comment]
@vrunoa For location mocking I'd rather create a separate class, which can send all supported commands via emulator console:
http://stackoverflow.com/questions/2279647/how-to-emulate-gps-location-in-the-android-emulator
https://developer.android.com/studio/run/emulator-console.html

Please let me know if any support from my side is needed there.

**Comment by vrunoa** [comment]
@mykola-mokhnach yeah, my idea was to add it in here [https://github.com/appium/appium-adb/blob/master/lib/tools/adb-emu-commands.js](https://github.com/appium/appium-adb/blob/master/lib/tools/adb-emu-commands.js)

**Comment by mykola-mokhnach** [comment]
Yep. There is also _sendTelnetCommand_ method in _tools/adb-commands.js_. You may try to reuse it.

**Comment by vrunoa** [comment]
@mykola-mokhnach we can use `adb emu geo fix 50 50`

**Comment by mykola-mokhnach** [comment]
yep, that's even simpler then

**Comment by mykola-mokhnach** [comment]
@vrunoa Added "adb emu geo fix" call. Please take a look now.
This will allow us to get rid of settings app installation on emulator

**Comment by vrunoa** [comment]
@mykola-mokhnach I have code ready to PR, but I would like to add some functional tests, do you guys have an app for testing over emulators ?

**Comment by mykola-mokhnach** [comment]
@vrunoa Functional tests are currently broken here. It's better to ask @imurchie about the current situation on Travis.
And regarding location mocking - I think it would be good to merge it in single PR, since then I can exclude settings app installation for simulator in android driver. @vrunoa Would you mind to merge your changes into this PR?

**Comment by vrunoa** [comment]
sure @mykola-mokhnach I'll add them tonight

File: lib/tools/adb-commands.js [hunk 2]
```diff
@@ -402,24 +408,34 @@ methods.isDataOn = async function () {
...
   }
   if (!_.isUndefined(data)) {
-    dataOpts = ['-e', 'data', (data ? 'on' : 'off')];
+    this.setDataState(data, isEmulator);
+  }
+};
+
+methods.setGeoLocation = async function (location, isEmulator = false) {
+  if (isEmulator) {
+    this.adbExec(['emu', 'geo', 'fix', ''+ location.longitude, '' + location.latitude]);
```
Comment (jlipps) [inline lib/tools/adb-commands.js]: template strings please!

Comment (jlipps) [inline lib/tools/adb-commands.js]: this is nice, we don't need telnet anymore? that's how we've been doing it but also it was flakey.

Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: This was a recommendation from @vrunoa. If you say it is not very stable to do it via adb them we can anytime change it to telnet if we get complains from users.

Comment (jlipps) [inline lib/tools/adb-commands.js]: No I think this is great! I just didn't know about it, and think it is something that didn't exist when we did the telnet option. Maybe this is much more stable.

Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: @jlipps I've checked what people say about that and found the bug
https://code.google.com/p/android/issues/detail?id=206180
Is there any method in Node to set decimal splitter based on location?

Comment (vrunoa) [inline lib/tools/adb-commands.js]: @jlipps @mykola-mokhnach  I tried this yesterday and I notice some weird issue, I could set some locations, for ex. I put my city and Vancouver and it didnt change, but it was working with Google example. I wonder if this was an issue with the Google Maps app or the emu. I would like to create an app to tests this before removing the .settings helpers app to set the mock location.

Comment (vrunoa) [inline lib/tools/adb-commands.js]: nice catch @mykola-mokhnach

Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: ok, I'll execute the command twice - once with dots and the second time with commas. Then it'll work in all cases even without knowing the actual locale.

**Comment by vrunoa** [comment]
@mykola-mokhnach can you add `this.resetTelnetAuthToken();` to the `emuMethods.fingerprint` as well ? That might also fail if the token was erased

**Comment by imurchie** [comment]
Published in `appium-adb@2.16.0`.

**Pull request merged**